src/lib/Liblog/pbs_log.c

```diff
@@ -931,6 +931,7 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 			     eventtype & ~PBSEVENT_FORCE, msg_daemonname,
 			     class_names[objclass], objname, text);
 
+		(void)fflush(logfile);
```

Comment:
This could have a significant performance impact in some cases. Would it not be better to make this configurable?

Reply:
HI @mike0042 -  the fflush existed in earlier code which we removed recently, without realizing that the called used were the buffered libc calls that would keep a log message in the application buffers (without the fflush) and if the application crashed, one might then lose critical/crucial log information that might point out what the application was doing during its last moments of life. This seriously hampers functionality. The gain in performance by removing the fflush is only seen in very slow filesystems like a home on NFS etc. but customers always have the option of going to syslog in those cases. Besides the performance gain (far we measured) was barely modest. Hence, we want to restore functionality.

Reply:
Hello @subhasisb - If we have numbers to backup the performance impact is minimal in nearly all normal cases of operation, I don't see the point of making it configurable. I withdraw my request, and thank you for the explanation.

Reply:
hi @mike0042  I will post the performance numbers in the PR by today